[pickers] Instructions for installing the package are not complete on the README

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

```
$ react-scripts start

./node_modules/@date-io/date-fns/build/index.esm.js
Module not found: Can't resolve 'date-fns/_lib/format/longFormatters'
```

### Expected behavior 🤔

Server goes fine.

### Steps to reproduce 🕹

Steps:

1. `npx create-react-app app`
2. `npm i @mui/material @mui/system @mui/x-date-pickers react react-dom`
3. npm start

### Context 🔦

I'm trying to follow the [installation guide of `@mui/x-date-pickers`](https://www.npmjs.com/package/@mui/x-date-pickers) that says these are the peer dependencies : 

```json
"peerDependencies": {
  "@mui/material": "^5.2.3",
  "@mui/system": "^5.2.3",
  "react": "^17.0.2 || ^18.0.0",
  "react-dom": "^17.0.2 || ^18.0.0"
},
```

But I think it's missing `date-fns`.


### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
System:
    OS: Linux 5.13 Ubuntu 20.04.4 LTS (Focal Fossa)
  Binaries:
    Node: 14.18.1 - ~/.nvm/versions/node/v14.18.1/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 6.14.15 - ~/.nvm/versions/node/v14.18.1/bin/npm
  Browsers:
    Chrome: Not Found
    Firefox: 99.0
  npmPackages:
    @emotion/react: ^11.9.0 => 11.9.0 
    @emotion/styled: ^11.8.1 => 11.8.1 
    @mui/base:  5.0.0-alpha.76 
    @mui/icons-material: ^5.6.2 => 5.6.2 
    @mui/material: ^5.5.3 => 5.6.1 
    @mui/private-theming:  5.7.0 
    @mui/styled-engine:  5.7.0 
    @mui/system: ^5.7.0 => 5.7.0 
    @mui/types:  7.1.3 
    @mui/utils:  5.6.1 
    @mui/x-date-pickers: ^5.0.0-alpha.2 => 5.0.0-alpha.2 
    @types/react:  17.0.43 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 

```
</details>


### Order ID 💳 (optional)

_No response_

--- a/docs/data/date-pickers/getting-started/getting-started-pt.md
+++ b/docs/data/date-pickers/getting-started/getting-started-pt.md
@@ -35,9 +35,9 @@ Um exemplo de seletor de data nativo com `type="date"`.
 // date-fns
 npm install @date-io/date-fns
 // or for Day.js
-npm install -s @date-io/dayjs
+npm install @date-io/dayjs
 // or for Luxon
-npm install -s @date-io/luxon
+npm install @date-io/luxon
 // or for Moment.js
 npm install @date-io/moment
 ```

--- a/docs/data/date-pickers/getting-started/getting-started-zh.md
+++ b/docs/data/date-pickers/getting-started/getting-started-zh.md
@@ -35,9 +35,9 @@ packageName: '@mui/x-date-pickers'
 // date-fns
 npm install @date-io/date-fns
 // or for Day.js
-npm install -s @date-io/dayjs
+npm install @date-io/dayjs
 // or for Luxon
-npm install -s @date-io/luxon
+npm install @date-io/luxon
 // or for Moment.js
 npm install @date-io/moment
 ```

--- a/docs/data/date-pickers/getting-started/getting-started.md
+++ b/docs/data/date-pickers/getting-started/getting-started.md
@@ -36,9 +36,9 @@ First you have to install the adapter package for the date-library you want to u
 // date-fns
 npm install @date-io/date-fns
 // or for Day.js
-npm install -s @date-io/dayjs
+npm install @date-io/dayjs
 // or for Luxon
-npm install -s @date-io/luxon
+npm install @date-io/luxon
 // or for Moment.js
 npm install @date-io/moment
 ```

--- a/docs/data/date-pickers/getting-started/getting-started.md
+++ b/docs/data/date-pickers/getting-started/getting-started.md
@@ -30,6 +30,8 @@ We currently support 4 different date-libraries:
 - [Luxon](https://moment.github.io/luxon/#/)
 - [Moment.js](https://momentjs.com/)
 
+If you need to use `js-joda`, `date-fns-jalali`, `jalaali`, or `hijri` library, you should be able to find the corresponding date-library from [`@date-io`](https://github.com/dmtrKovalenko/date-io#projects).
+
 First you have to install the adapter package for the date-library you want to use:
 
 ```sh

--- a/packages/x-date-pickers-pro/README.md
+++ b/packages/x-date-pickers-pro/README.md
@@ -26,6 +26,49 @@ This component has the following peer dependencies that you will need to install
 },
 ```
 
+You need to provide a date-library that is used by the pickers by setting the `dateAdapter` to an adapter of your choosing.
+
+We currently support 4 different date-libraries:
+
+- [date-fns](https://date-fns.org/)
+- [Day.js](https://day.js.org/)
+- [Luxon](https://moment.github.io/luxon/#/)
+- [Moment.js](https://momentjs.com/)
+
+First, you have to install the adapter package for the date-library you want to use:
+
+```sh
+// date-fns
+npm install @date-io/date-fns
+// or for Day.js
+npm install -s @date-io/dayjs
+// or for Luxon
+npm install -s @date-io/luxon
+// or for Moment.js
+npm install @date-io/moment
+```
+
+Then you have to set the `dateAdapter` prop of the `LocalizationProvider` accordingly:
+
+```js
+// date-fns
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+// or for Day.js
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+// or for Luxon
+import { AdapterLuxon } from '@mui/x-date-pickers/AdapterLuxon';
+// or for Moment.js
+import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
+
+function App({ children }) {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      {children}
+    </LocalizationProvider>
+  );
+}
+```
+
 ## Documentation
 
 [The documentation](https://mui.com/x/react-date-pickers/getting-started/)

--- a/packages/x-date-pickers-pro/README.md
+++ b/packages/x-date-pickers-pro/README.md
@@ -35,15 +35,17 @@ We currently support 4 different date-libraries:
 - [Luxon](https://moment.github.io/luxon/#/)
 - [Moment.js](https://momentjs.com/)
 
+If you need to use `js-joda`, `date-fns-jalali`, `jalaali`, or `hijri` library, you should be able to find the corresponding date-library from [`@date-io`](https://github.com/dmtrKovalenko/date-io#projects).
+
 First, you have to install the adapter package for the date-library you want to use:
 
 ```sh
 // date-fns
 npm install @date-io/date-fns
 // or for Day.js
-npm install -s @date-io/dayjs
+npm install @date-io/dayjs
 // or for Luxon
-npm install -s @date-io/luxon
+npm install @date-io/luxon
 // or for Moment.js
 npm install @date-io/moment
 ```

--- a/packages/x-date-pickers-pro/README.md
+++ b/packages/x-date-pickers-pro/README.md
@@ -61,11 +61,7 @@ import { AdapterLuxon } from '@mui/x-date-pickers/AdapterLuxon';
 import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
 
 function App({ children }) {
-  return (
-    <LocalizationProvider dateAdapter={AdapterDateFns}>
-      {children}
-    </LocalizationProvider>
-  );
+  return <LocalizationProvider dateAdapter={AdapterDateFns}>{children}</LocalizationProvider>;
 }
 ```
 

--- a/packages/x-date-pickers/README.md
+++ b/packages/x-date-pickers/README.md
@@ -26,6 +26,49 @@ This component has the following peer dependencies that you will need to install
 },
 ```
 
+You need to provide a date-library that is used by the pickers by setting the `dateAdapter` to an adapter of your choosing.
+
+We currently support 4 different date-libraries:
+
+- [date-fns](https://date-fns.org/)
+- [Day.js](https://day.js.org/)
+- [Luxon](https://moment.github.io/luxon/#/)
+- [Moment.js](https://momentjs.com/)
+
+First, you have to install the adapter package for the date-library you want to use:
+
+```sh
+// date-fns
+npm install @date-io/date-fns
+// or for Day.js
+npm install -s @date-io/dayjs
+// or for Luxon
+npm install -s @date-io/luxon
+// or for Moment.js
+npm install @date-io/moment
+```
+
+Then you have to set the `dateAdapter` prop of the `LocalizationProvider` accordingly:
+
+```js
+// date-fns
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+// or for Day.js
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+// or for Luxon
+import { AdapterLuxon } from '@mui/x-date-pickers/AdapterLuxon';
+// or for Moment.js
+import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
+
+function App({ children }) {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      {children}
+    </LocalizationProvider>
+  );
+}
+```
+
 ## Documentation
 
 [The documentation](https://mui.com/x/react-date-pickers/getting-started/)

--- a/packages/x-date-pickers/README.md
+++ b/packages/x-date-pickers/README.md
@@ -35,15 +35,17 @@ We currently support 4 different date-libraries:
 - [Luxon](https://moment.github.io/luxon/#/)
 - [Moment.js](https://momentjs.com/)
 
+If you need to use `js-joda`, `date-fns-jalali`, `jalaali`, or `hijri` library, you should be able to find the corresponding date-library from [`@date-io`](https://github.com/dmtrKovalenko/date-io#projects).
+
 First, you have to install the adapter package for the date-library you want to use:
 
 ```sh
 // date-fns
 npm install @date-io/date-fns
 // or for Day.js
-npm install -s @date-io/dayjs
+npm install @date-io/dayjs
 // or for Luxon
-npm install -s @date-io/luxon
+npm install @date-io/luxon
 // or for Moment.js
 npm install @date-io/moment
 ```

--- a/packages/x-date-pickers/README.md
+++ b/packages/x-date-pickers/README.md
@@ -61,11 +61,7 @@ import { AdapterLuxon } from '@mui/x-date-pickers/AdapterLuxon';
 import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
 
 function App({ children }) {
-  return (
-    <LocalizationProvider dateAdapter={AdapterDateFns}>
-      {children}
-    </LocalizationProvider>
-  );
+  return <LocalizationProvider dateAdapter={AdapterDateFns}>{children}</LocalizationProvider>;
 }
 ```
 

